subprojects/build-cache-http/src/integTest/groovy/org/gradle/caching/http/internal/HttpBuildCacheServiceTest.groovy

```diff
@@ -293,6 +298,36 @@ class HttpBuildCacheServiceTest extends Specification {
         server.authenticationAttempts == ['Basic'] as Set
     }
 
+    @Issue("gradle/gradle#5571")
+    def "does preemptive authentication authenticating with http header"() {
+        def configuration = new HttpBuildCache(new HttpBuildCacheHttpHeadCredentials("HeaderName: HeaderValue"))
+        configuration.url = server.uri.resolve("/cache/")
+        cache = new DefaultHttpBuildCacheServiceFactory(new DefaultSslContextFactory()).createBuildCacheService(configuration, buildCacheDescriber) as HttpBuildCacheService
+
+        server.authenticationScheme = AuthScheme.HEADER
+
+        def destFile = tempDir.file("cached.zip")
+        destFile.text = 'Old'
+        when:
+        server.expectGet("/cache/${key.hashCode}", destFile)
+        def result = null
+        cache.load(key) { input ->
+            result = input.text
+        }
+        then:
+        result == 'Old'
+        server.authenticationAttempts == ['Header'] as Set
+
+        server.expectPut("/cache/${key.hashCode}", destFile)
+
+        when:
+        def content = "Data".bytes
+        cache.store(key, writer(content))
+        then:
+        destFile.bytes == content
+        server.authenticationAttempts == ['Header'] as Set
+    }
+
```

Comment:
Why also support the build cache? I thought #5571 was about Maven repositories?

Reply:
Yes, please remove all changes to `HttpBuildCache*` classes from this PR. If we want to support header-based authentication for the build cache, we'll do it in a separate PR.

Reply:
Ok, Done.